UCLALibrary/libguides-indexing-scripts
# SYS-1172: Prepare for production deployment

Implements changes supporting [SYS-1172](https://jira.library.ucla.edu/browse/SYS-1172).

This PR adds a shell script, `update_libguides_index.sh`, which should be suitable for running on a server via cron.
It runs the 3rd-party `libguider.py` program to download our LibGuides as HTML files, then runs our `libguides_elastic_client.py` to extract content and index it in Elasticsearch.

The shell script relies on a configuration file, `libguides_secrets.cfg`, which is not included in this repo as it contains sensitive information.  That file, and the `api_keys.py` we're already using for ES, need to be copied and installed manually into the same directory as `update_libguides_index.sh` and `libguides_elastic_client.py` .

The shell script requires one argument: one of `TEST`, `STAGE`, or `PROD`, indicating which index to rebuild.  The actual index names are pulled from the configuration file.

This PR also:
* Updates `elastic_lib_client.py` with comment noting that the target ES index must be referenced explicitly.
* Makes our script use explicit ES index names throughout.
* Renames `test_elastic_client.py` to `libguides_elastic_client.py`
* Adds an optional `-f / --file_spec` argument to `libguides_elastic_client.py`, to allow indexing of only specific files when testing.
* Deletes documents from ES index before rebuilding it.

### Testing
* Add `libguides_secrets.cfg` and current `api_keys.py` to local installation
* Make sure `pipenv` is installed (should be already, as these repos both use that)
* Connect to UCLA VPN, for access to our ES index server
* Run `update_libguides_index.sh`
